transcriber.py

```diff
@@ -75,6 +76,10 @@ def add(
                               filename[:-4])
     else:
         # process video or a playlist
+        if source.startswith("https://www.youtube.com"):
```

Comment:
![Screen Shot 2023-02-01 at 8 56 21 PM](https://user-images.githubusercontent.com/34213201/216212180-0ae1fcaa-28bd-4868-9223-13166d8b59fc.png)


@masterchief164 - heads up, I think this ultimately works just fine, but it will miss youtube links that don't begin with "https". For example, I tried running a transcript for a link that began with "youtube.com", and it appended it to the end of "https://www.youtube.com/watch?v=" instead of extracting out the id (see top picture).

However, the link is ultimately passed to pytube, which seems to be **_very_** robust to all forms of youtube links. It even works for blatant misspellings ("you.com/watch?v=EcFbDC9Jzuo"), which is kinda cool.

Reply:
We will need a starting point for identifying if the link is a youtube link or not. I think I can add a check that looks for a `youtube` or more keywords that might be provided by youtube like `you.tube` and then go forward with that else it will assume that the source is a playlist or a video id.

Reply:
Okay, great. That sounds like it will work.

Also, on further expectation, it looks like the [check_if_playlist](https://github.com/bitcointranscripts/tbtbtc/blob/aaf7ba35cd0d15d77107cca1cc20e00a3cdd540c/app/application.py#L111) function expects video IDs, so I believe my example above would have been missed (assuming it was a playlist) because the source URL would have been passed to check_if_playlist instead of the video ID.

Perhaps, the easiest way to check for youtube links is by using `pytube`, since it seems to do a great job of checking if the link is actually a youtube video. I've included some example code below for what that function could look like.

```
def is_youtube_video(url):
    try:
        video = pytube.YouTube(url)
        return True
    except:
        return False
```

Reply:
I think a better approach would be to use a regex. Infact we could use the code from pytube itself.